File: Magic.Domain/Entities/TileProperty.cs
using Magic.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Magic.Domain.Entities
{
    public class TileProperty : BaseEntity<int>
    {
        /// <summary>
        /// Название тайла
        /// </summary>
        public string Title { get; set; }
        /// <summary>
        /// Описание тайла
        /// </summary>
        public string Description { get; set; }
        /// <summary>
        /// Изображение тайла
        /// </summary>
        public string Image {  get; set; }
        /// <summary>
        /// Тип колизии тайла
        /// </summary>
        public TilePropertyCollisionTypeEnum CollisionType { get; set; }
        /// <summary>
        /// Тип штрафа за проход по тайлу
        /// </summary>
        public TilePropertyPenaltyTypeEnum PenaltyType { get; set; }
        /// <summary>
        /// Тип возможного действия по отношению к тайлу
        /// </summary>
        public TilePropertyTargetTypeEnum TargetType { get; set; }
        /// <summary>
        /// Количество прочности у тайла
        /// </summary>
        public int? Health { get; set; }
        /// <summary>
        /// Идентификатор тайла, который должен появится при разрушении
        /// </summary>
        public int? TilePropertyIdIfDestroyed { get; set; }
        /// <summary>
        /// Тайл, который должен появится при разрушении
        /// </summary>
        public TileProperty TilePropertyIfDestroyed { get; set; }
        /// <summary>
        /// Количество штрафа за проход по клетке
        /// </summary>
        public int? PenaltyValue { get; set; }
    }
}

File: Magic.Frontend/src/models/response/tileProperty.ts
import {BaseEntity} from "./characterTemplateResponse";

export interface TileProperty extends BaseEntity<number> {
    title: string;
    description: string;
    image: string;
    collisionType: TilePropertyCollisionTypeEnum;
    penaltyType: TilePropertyPenaltyTypeEnum;
    targetType: TilePropertyTargetTypeEnum;
    health: number | null;
    tilePropertyIdIfDestroyed: number | null;
    tilePropertyIfDestroyed: TileProperty | null;
    penaltyValue: number | null;
}

export enum TilePropertyCollisionTypeEnum {
    /// Свободный проход по клетке
    None = 1,
    /// Движение через этот тайл невозможно
    NoMove = 2,
    /// Движение только с полетом
    OnlyFly = 3,
}

export enum TilePropertyPenaltyTypeEnum {
    /// Нет штрафа за проход по клетке
    None = 1,
    /// Штраф в виде жизней
    PenaltyHealth = 2,
    /// Штраф в виде скорости
    PenaltySpeed = 3,
}

export enum TilePropertyTargetTypeEnum {
    /// Тайл нельзя выбрать в таргет
    None = 1,
    /// Разрушаемый
    Destructible = 2,
}
